src/ucp/core/ucp_context.c

```diff
@@ -126,11 +126,21 @@ static ucs_config_field_t ucp_config_table[] = {
    "y      - Use mutex for multithreading support in UCP.\n",
    ucs_offsetof(ucp_config_t, ctx.use_mt_mutex), UCS_CONFIG_TYPE_BOOL},
 
+  {"SEG_SIZE", "8192",
+   "Size of a segment in the worker preregistered memory pool.",
+   ucs_offsetof(ucp_config_t, ctx.seg_size), UCS_CONFIG_TYPE_MEMUNITS},
+
   {"TM_THRESH", "1024",
    "Threshold for using tag matching offload capabilities.\n"
    "Smaller buffers will not be posted to the transport.",
    ucs_offsetof(ucp_config_t, ctx.tm_thresh), UCS_CONFIG_TYPE_MEMUNITS},
 
+  {"TM_MAX_BCOPY", "0",
+   "Maximal size for posting \"bounce buffer\" (UCX interal preregistered memory) for\n"
+   "tag offload receives. When message arrives, it is copied into the user buffer (similar\n"
+   "to eager protocol). The size is limited by worker segment size.",
```

Comment:
The size values has to be equal or less than segment size.

I would add early stage configure check that reports an error if user sets "bad" value